pkg/util/detector/detector.go

```diff
@@ -626,7 +626,8 @@ func (d *ResourceDetector) BuildResourceBinding(object *unstructured.Unstructure
 			OwnerReferences: []metav1.OwnerReference{
 				*metav1.NewControllerRef(object, objectKey.GroupVersionKind()),
 			},
-			Labels: labels,
+			Labels:     labels,
+			Finalizers: []string{util.BindingControllerFinalizer},
```

Comment:
I guess we should update this place as well:
https://github.com/karmada-io/karmada/blob/1947cf219979c728f1aa63fd185b33efd821b1de/pkg/util/detector/detector.go#L438-L445

Same as below.

Reply:
+1, bindingCopy.Finalizers = binding.Finalizers

Reply:
I'm not sure why updating finalizer here, this mutate function is for updating but it already has finalizer when it was created, when do we need to reset finalizer?